hub/core/transform/transform_tensor.py

```diff
@@ -0,0 +1,59 @@
+from hub.core.sample import Sample  # type: ignore
+from hub.util.exceptions import TensorInvalidSampleShapeError
+import numpy as np
+
+
+class TransformDatasetTensor:
+    def __init__(self, base_tensor=None, slice_list=None) -> None:
+        self.items = [] if base_tensor is None else base_tensor.items
+        self.base_tensor = base_tensor or self
+        self.slice_list = slice_list or []
+        self.length = None
+
+    def numpy(self) -> None:
+        value = self.numpy_compressed()
+        if isinstance(value, list):
+            for i in range(len(value)):
+                if isinstance(value[i], Sample):
+                    value[i] = value[i].array
+        elif isinstance(value, Sample):
+            value = value.array
+        return value
+
+    def numpy_compressed(self):
+        value = self.items
+        for slice_ in self.slice_list:
+            value = value[slice_]
+        self.length = len(value) if isinstance(value, list) else 1
+        return value
```

Comment:
what does this function do? why is it `numpy_compressed`?

Reply:
when user calls .numpy() it should return numpy values and in case of hub.read being used, the sample gets converted to numpy array. When I'm sending the data to chunk_engine, I instead call numpy_compressed to not convert the sample into numpy array, so that we can maintain the hub.read speedup.